pkg/client/unversioned/conditions.go

```diff
@@ -168,3 +171,125 @@ func DeploymentHasDesiredReplicas(c ExtensionsInterface, deployment *extensions.
 			deployment.Status.UpdatedReplicas == deployment.Spec.Replicas, nil
 	}
 }
+
+// ErrPodCompleted is returned by PodRunning or PodContainerRunning to indicate that
+// the pod has already reached completed state.
+var ErrPodCompleted = fmt.Errorf("pod ran to completion")
```

Comment:
Most of the code that benefits from using this the most now is not typed client.  I would like to bake these a bit before adding them to the typed client, because a) we may find different conditions or b) want to structure them differently after use.  e2e, server code, and controllers.  I would expect someone converting code that uses this to typed clients would be the one who does the conversion once they've settled enough.